R/redundant_ifelse_linter.R

```diff
@@ -0,0 +1,62 @@
+#' Prevent ifelse() from being used to produce TRUE/FALSE or 1/0
+#'
+#' Expressions like `ifelse(x, TRUE, FALSE)` and `ifelse(x, FALSE, TRUE)` are
```

Comment:
I think `ifelse(cond, 1.0, 0.0)` is fine actually, because it doesn't require the reader to know that `as.numeric(cond)` has the same result.

WDYT about making the numeric coercion lint configurable?

Reply:
SGTM; done.

Leaving linting on by default. I do think `ifelse(cond, 1, 0)` is still usually the wrong thing to do.